frontend/src/metabase/actions/components/ActionViz/Action.tsx

```diff
@@ -120,20 +123,18 @@ const ConnectedActionComponent = connect()(ActionComponent);
 function mapStateToProps(state: State, props: ActionProps) {
   return {
     isEditingDashcard: getEditingDashcardId(state) === props.dashcard.id,
-    metadata: getMetadata(state),
   };
 }
 
 export function ActionFn(props: ActionProps) {
   const {
-    metadata,
+    database,
     dashcard: { action },
   } = props;
-  const actionsEnabled = !!metadata
-    ?.database(action?.database_id)
-    ?.hasActionsEnabled?.();
 
-  if (!props.dashcard?.action || !actionsEnabled) {
+  const hasActionsEnabled = database?.hasActionsEnabled?.();
```

Comment:
Entity loader will not render the component if it cannot load the entity. I think you can assume it will be always defined, e.g. just `database.hasActionsEnabled()`

Reply:
🤔  I don't think that's the actual or intended behavior here. We definitely want to render in the absence of a DB so we can show an "empty" button that needs an action assigned.

![Screen Shot 2023-02-20 at 8 08 12 AM](https://user-images.githubusercontent.com/30528226/220144992-482c7e0a-c8fa-4776-8ec3-28f144a18019.png)